### PR TITLE
 Add JSON code generation

### PR DESCRIPTION
**Pull request by tvh**

### CODE REVIEW

File: hprotoc/Text/ProtocolBuffers/ProtoCompile/Gen.hs [hunk 1]
```diff
@@ -786,6 +807,96 @@ instanceUnknownMessage di
...
+        flds = F.toList (fields di)
+        msgVar = distinctVar "msg"
+        reservedVars = map toPrintName flds
+        distinctVar var = if var `elem` reservedVars then distinctVar (var ++ "'") else var
+        getFname fld = fName $ baseName' $ fieldName fld
+        makePair fld =
+            let fldName = getFname fld
+                fldName' = dropWhileEnd (== '\'') fldName
+                toJSONFun = case typeCode fld of
+                    3 -> pvar "toJSONShowWithPayload"
```
Comment (skogsbaer) [inline hprotoc/Text/ProtocolBuffers/ProtoCompile/Gen.hs]: Gibt's für diese Zahlen keine symbolischen Konstanten??

Comment (tvh) [reply hprotoc/Text/ProtocolBuffers/ProtoCompile/Gen.hs]: Yes, but I thought I couldn't use them. There is http://hackage.haskell.org/package/protocol-buffers-descriptor-2.4.10/docs/Text-DescriptorProtos-FieldDescriptorProto-Type.html#t:Type, but that would mean parsing it first.

Comment (skogsbaer) [reply hprotoc/Text/ProtocolBuffers/ProtoCompile/Gen.hs]: what about introducing symbolic constants?

Comment (tvh) [reply hprotoc/Text/ProtocolBuffers/ProtoCompile/Gen.hs]: I just added the import. The package was in scope anyway. Not sure what this does for bootstrapping though.

**Review by skogsbaer — APPROVED** [review]

**Pull request merged**